src/modules/systemlib/battery.cpp

```diff
@@ -164,15 +164,15 @@ Battery::sumDischarged(hrt_abstime timestamp, float current_a)
 }
 
 void
-Battery::estimateRemaining(float voltage_v, float current_a, float throttle_normalized, bool armed)
+Battery::estimateRemaining(float throttle_normalized, bool armed)
 {
 	const float bat_r = _param_r_internal.get();
 
 	// remaining charge estimate based on voltage and internal resistance (drop under load)
 	float bat_v_empty_dynamic = _param_v_empty.get();
 
 	if (bat_r >= 0.0f) {
-		bat_v_empty_dynamic -= current_a * bat_r;
+		bat_v_empty_dynamic -= _current_filtered_a * bat_r;
```

Comment:
I think you want to integrate the non-filtered current to be more accurate. And integrating is in fact filtering in that respect.

Reply:
this is not an integration operation, but rather a correction for voltage drop. This value gets reset during every update [here](https://github.com/PX4/Firmware/blob/master/src/modules/systemlib/battery.cpp#L172) and is then used here to compute the remaining voltage [here](https://github.com/PX4/Firmware/blob/master/src/modules/systemlib/battery.cpp#L189). Since in this PR rvoltage uses the filtered voltage [here](https://github.com/PX4/Firmware/blob/master/src/modules/systemlib/battery.cpp#L189), so I think it makes sense to alos use the filtered current to compute the correction term bat_v_empty_dynamic,
However, I still have the question about filtering the remaining capacitance [here](https://github.com/PX4/Firmware/blob/master/src/modules/systemlib/battery.cpp#L198-L199), because _discharged_mah is indeed integrated [here](https://github.com/PX4/Firmware/blob/master/src/modules/systemlib/battery.cpp#L160)

Reply:
Ok, I misunderstood this. I think you're right. And it looks like where `_discharged_mah` is integrated, it uses the non-filtered value?

Reply:
yes, `_discharged_mah` uses the non-filtered current for integration. Do you agree about removing filter for remaining capacitance [here](https://github.com/PX4/Firmware/blob/master/src/modules/systemlib/battery.cpp#L199) since it uses the `_discharged_mah`?

Reply:
Right, I don't understand why it should be filtered there.